pkg/deploy/lattice/listener_synthesizer.go

```diff
@@ -65,9 +66,8 @@ func (l *listenerSynthesizer) Synthesize(ctx context.Context) error {
 
 		glog.V(6).Infof("ListenerSynthesize >>>> delete staled sdk listener %v\n", *sdkListener)
 		l.listener.Delete(ctx, sdkListener.ListenerID, sdkListener.ServiceID)
-		k8sname, k8snamespace := latticeName2k8s(sdkListener.Name)
 
-		l.latticestore.DelListener(k8sname, k8snamespace, sdkListener.Port, sdkListener.Protocol)
+		l.latticestore.DelListener(sdkListener.Name, sdkListener.Namespace, sdkListener.Port, sdkListener.Protocol)
```

Comment:
I do not think sdkListener.Namespace is populated by l.getSDKListeners().  Not sure why unit test did not catch this? Also, i think the e2e should catch this too. (not sure why )

Reply:
can u confirm this from the debugging output at line#58

Reply:
Also,  for all these xxxSDKxxx() routine(s), I think we should use the name or tags return from lattice to figure out the namespace.

Reply:
I made the change to that function just right below actually, I think it is working?

On listener creation:
```
I0519 15:20:51.334060   39938 listener_synthesizer.go:126] getSDKListeners result >> [0xc00090cf50]
I0519 15:20:51.334069   39938 listener_synthesizer.go:58] getSDKlistener: [0xc00090cf50], err: <nil>
I0519 15:20:51.334074   39938 listener_synthesizer.go:86] findMatchListener: found matching moduleListener &{{0xc000a59280 AWS::VPCServiceNetwork::Listener inventory-non-default-80-HTTP} {inventory non-default 80 HTTP {false inventory-ver1 non-default}} <nil>}
```

Reply:
The objective of these xxxSDKxxx() is to figure out if Lattice resource(s) can map back to K8S resource.  It should `not` imply nor inherent any value from K8S object.

Reply:
It is impossible to figure out exact namespace because of the naming scheme of lattice and k8s. Lets say we have lattice listener `a-b-c-d-e-80-HTTP`, the resource can be either one of:
`a/b-c-d-e`
`a-b/c-d-e`
`a-b-c/d-e`
`a-b-c-d/e`
I think we need to rely on k8s information to figure out namespace. it should be the same as the lattice service anyways.

Reply:
then we need to explicit using `tag`.  The whole idea is to compare between lattice resources and K8S object(s).  And perform cleanup lattice resources if they are not matching K8S objects.  

If we start infer lattice resource(s) from K8S object(s), we will end up leaking more lattice resource(s)